src/mutate/drawDefinitions/positionGovernor/qualifierProgression.ts

```diff
@@ -0,0 +1,163 @@
+import { findExtension } from '@Acquire/findExtension';
+import { findStructure } from '@Acquire/findStructure';
+import { qualifierDrawPositionAssignment } from '@Assemblies/governors/drawsGovernor';
+import { POSITION, QUALIFYING, WINNER } from '@Constants/drawDefinitionConstants';
+import {
+  MISSING_QUALIFIED_PARTICIPANTS,
+  NO_DRAW_POSITIONS_AVAILABLE_FOR_QUALIFIERS,
+} from '@Constants/errorConditionConstants';
+import { TALLY } from '@Constants/extensionConstants';
+import { BYE } from '@Constants/matchUpStatusConstants';
+import { POLICY_TYPE_POSITION_ACTIONS } from '@Constants/policyConstants';
+import { SUCCESS } from '@Constants/resultConstants';
+import { decorateResult } from '@Functions/global/decorateResult';
+import { getPositionAssignments, structureAssignedDrawPositions } from '@Query/drawDefinition/positionsGetter';
+import { isCompletedStructure } from '@Query/drawDefinition/structureActions';
+import { getAppliedPolicies } from '@Query/extensions/getAppliedPolicies';
+import { getAllStructureMatchUps } from '@Query/matchUps/getAllStructureMatchUps';
+import { getSourceStructureIdsAndRelevantLinks } from '@Query/structure/getSourceStructureIdsAndRelevantLinks';
+import { randomPop } from '@Tools/arrays';
+import { definedAttributes } from '@Tools/definedAttributes';
+import { ResultType } from '@Types/factoryTypes';
+import { DrawDefinition, Event, Structure, Tournament } from '@Types/tournamentTypes';
+
+interface QualifierProgressionArgs {
+  drawDefinition: DrawDefinition;
+  event: Event;
+  mainStructureId: string;
+  tournamentRecord: Tournament;
+}
+
+export function qualifierProgression({
+  drawDefinition,
+  event,
+  mainStructureId,
+  tournamentRecord,
+}: QualifierProgressionArgs): ResultType {
+  const qualifyingParticipantIds: string[] = [];
+
+  const structure = findStructure({ drawDefinition, structureId: mainStructureId })?.structure ?? ({} as Structure);
```

Comment:
I'd be more comfortable if this were explicitly `mainStructure`

Reply:
Also, can we trust that the value passed in as `mainStructureI` is actually the `structure` id of MAIN?  I think we should check that the found structure.stage === MAIN and structure.stageSequence === 1 or throw an error.

And, why even require the mainstructureId to be passed in?  Since it is always the only structure we're interested in, why not just derive it from the drawDefinition since we want to check anyway?